src/catchup/ApplyBucketsWork.h

```diff
@@ -46,6 +46,27 @@ class ApplyBucketsWork : public BasicWork
     medida::Meter& mBucketApplyFailure;
     BucketApplicator::Counters mCounters;
 
+    // With protocol 12 or higher, when buckets are applied, we do not have
```

Comment:
There is something weird here that I think is not consistent with the behavior that you're describing:
if we're catching up but not applying buckets, we should probably wait for merges to be completed before we're claiming to be "in sync".

So... what I suspect is that we need to wait for merges to be completed in more places:
1. when running `ApplyLedgerChainWork` - maybe even between ledgers?
2. after applying each ledger when replaying buffered ledgers

This way we will guarantee that we're not blocking the main thread because of pending merges.

Reply:
NB: maybe this whole "wait to be in sync" functionality should be in its own commit as it seems non trivial (and independent from the other changes)